apiserver/facades/agent/storageprovisioner/storageprovisioner.go

```diff
@@ -97,6 +82,12 @@ func NewStorageProvisionerAPIv4(
 	}
 	controllerUUID := controllerConfig.ControllerUUID()
 
+	modelInfo, err := modelInfoService.GetModelInfo(ctx)
```

Comment:
We call this service method to get the modelUUID many times in the facade.
We can get the modelUUID in the register.go and cache in the `StorageProvisionerAPIv4`. You can get it from the facade context, so you even do not need `modelInfoService`.

Reply:
Good point, I've done this now.